packages/wallets/talisman/src/helpers.ts

```diff
@@ -0,0 +1,154 @@
+  return convertedAddress;
+};
+
+export const getWeb3WalletMethods = async ({
+  ethereumWindowProvider,
+  chain,
+  covalentApiKey,
+  ethplorerApiKey,
+}: {
+  ethereumWindowProvider: Eip1193Provider | undefined;
+  chain: Chain;
+  covalentApiKey?: string;
+  ethplorerApiKey?: string;
+}) => {
+  if (!ethereumWindowProvider) throw new Error("Requested web3 wallet is not installed");
+
+  if (
+    (chain !== Chain.Ethereum && !covalentApiKey) ||
+    (chain === Chain.Ethereum && !ethplorerApiKey)
+  ) {
+    throw new Error(`Missing API key for ${chain} chain`);
+  }
+
+  const provider = new BrowserProvider(ethereumWindowProvider, "any");
+
+  const toolboxParams = {
+    provider,
+    signer: await provider.getSigner(),
+    ethplorerApiKey: ethplorerApiKey as string,
+    covalentApiKey: covalentApiKey as string,
+  };
+
+  const toolbox =
+    chain === Chain.Ethereum
+      ? ETHToolbox(toolboxParams)
+      : chain === Chain.Arbitrum
+        ? ARBToolbox(toolboxParams)
+        : chain === Chain.Polygon
+          ? MATICToolbox(toolboxParams)
+          : chain === Chain.Optimism
+            ? OPToolbox(toolboxParams)
+            : BSCToolbox(toolboxParams);
+
+  try {
+    chain !== Chain.Ethereum &&
+      (await addEVMWalletNetwork(
+        provider,
+        (
+          toolbox as
+            | ReturnType<typeof ARBToolbox>
+            | ReturnType<typeof BSCToolbox>
+            | ReturnType<typeof MATICToolbox>
+            | ReturnType<typeof OPToolbox>
+        ).getNetworkParams(),
+      ));
+  } catch (_error) {
+    throw new Error(`Failed to add/switch ${chain} network: ${chain}`);
+  }
+
+  return prepareNetworkSwitch<typeof toolbox>({
+    toolbox: { ...toolbox },
+    chainId: ChainToHexChainId[chain],
+    provider,
+  });
+};
+
+export const getWalletForChain = async ({
+  chain,
+  ethplorerApiKey,
+  covalentApiKey,
+}: {
+  chain: Chain;
+  ethplorerApiKey?: string;
+  covalentApiKey?: string;
+}) => {
+  switch (chain) {
+    case Chain.Ethereum:
+    case Chain.Arbitrum:
+    case Chain.Optimism:
+    case Chain.Polygon:
+    case Chain.BinanceSmartChain: {
+      if (!(window.talismanEth && "send" in window.talismanEth)) {
+        throw new Error("No talisman wallet found");
+      }
+
+      const { getProvider } = await import("@swapkit/toolbox-evm");
+
+      const evmWallet = await getWeb3WalletMethods({
+        chain,
+        ethplorerApiKey,
+        covalentApiKey,
+        ethereumWindowProvider: window.talismanEth,
+      });
+
+      const address: string = (await window.talismanEth.send("eth_requestAccounts", []))[0];
+
+      const getBalance = async (addressOverwrite?: string, potentialScamFilter = true) =>
+        evmWallet.getBalance(addressOverwrite || address, potentialScamFilter, getProvider(chain));
+
+      return { walletMethods: { ...evmWallet, getBalance }, address };
+    }
+
+    case Chain.Polkadot: {
+      const injectedWindow = window as Window & InjectedWindow;
```

Comment:
it seems like it could be typed at "global" lvl as `declare window` 🤔 rest of wallets use this kind of typing - could we use it there as well (for polkadot I assume it would need `injectedWeb3: { talisman: ... }` typing

Reply:
fixed